crates/proto/src/op/header.rs

```diff
@@ -174,10 +174,15 @@ impl Header {
 
     /// The low response code (original response codes before EDNS extensions)
     pub fn set_response_code(&mut self, response_code: ResponseCode) -> &mut Self {
-        self.response_code = response_code.low();
+        self.response_code = response_code;
         self
     }
 
+    /// This combines the high and low response code values to form the complete ResponseCode from the EDNS record
```

Comment:
The semantics here seem a bit tricky. Maybe expand the comment to explain that this takes the low bits from `self` and the high bits from the argument?

Reply:
I've added additional docs, but given it's use is pretty much only for decoding, I think I'm going to mark it doc(hidden), thoughts?